app/workloadSummaryReport/services/workloadSummaryActions.js

```diff
@@ -0,0 +1,562 @@
+							assignment.description = "Buyout";
+						} else if (teachingAssignment.courseRelease) {
+							assignment.description = "Course Release";
+						} else if (teachingAssignment.sabbatical) {
+							assignment.description = "Sabbatical";
+						} else if (teachingAssignment.inResidence) {
+							assignment.description = "In Residence";
+						} else if (teachingAssignment.workLifeBalance) {
+							assignment.description = "Work Life Balance";
+						} else if (teachingAssignment.leaveOfAbsence) {
+							assignment.description = "Leave of Absence";
+						}
+
+						if (teachingAssignment.sectionGroupId > 0) {
+							var sectionGroup = sectionGroups.list[teachingAssignment.sectionGroupId];
+							var course = courses.list[sectionGroup.courseId];
+
+							assignment.description = course.subjectCode + " " + course.courseNumber;
+							assignment.sequencePattern = course.sequencePattern;
+							assignment.enrollment = _self._getEnrollment(sectionGroup);
+							assignment.previousEnrollment = sectionGroup.previousEnrollment;
+							assignment.units = _self._getUnits(course);
+							assignment.studentCreditHours = assignment.enrollment * assignment.units;
+						}
+
+						instructor.assignments.push(assignment);
+
+						instructor.totals.units += assignment.units || 0;
+						instructor.totals.studentCreditHours += assignment.studentCreditHours || 0;
+						instructor.totals.enrollment += assignment.enrollment || 0;
+						instructor.totals.previousEnrollment += assignment.previousEnrollment || 0;
+						instructor.totals.assignmentCount += 1;
+					});
+
+					calculatedView.byInstructorType[instructorTypeId].push(instructor);
+				});
+
+				calculatedView.instructorTypeIds = _self._orderInstructorTypeIdsAlphabetically(calculatedView.instructorTypeIds);
+
+				WorkloadSummaryReducers.reduce({
+					type: ActionTypes.CALCULATE_VIEW,
+					payload: {
+						calculatedView: calculatedView
+					}
+				});
+			},
+			_getUnits: function (course) {
+				if (course.unitsLow > 0) {
+					return course.unitsLow;
+				} else if (course.unitsHigh > 0) {
+					return course.unitsHigh;
+				}
+
+				return 0;
+			},
+			// Return actual (census), seats, or plannedSeats for the enrollment number, depending on what is available
+			_getEnrollment: function (sectionGroup) {
+				if (sectionGroup.actualEnrollment > 0) {
+					return sectionGroup.actualEnrollment ;
+				} else if (sectionGroup.maxEnrollment > 0) {
+					return sectionGroup.maxEnrollment;
+				} else {
+					return sectionGroup.plannedSeats;
+				}
+			},
+			_getInstructorTypeId: function (instructor) {
+				var teachingAssignments = WorkloadSummaryReducers._state.teachingAssignments;
+				var users = WorkloadSummaryReducers._state.users;
+				var userRoles = WorkloadSummaryReducers._state.userRoles;
+
+				var user = this._getUserByLoginId(instructor.loginId, users);
+
+				// Attempt to find via userRole
+				for (var i = 0; i < userRoles.ids.length; i++) {
```

Comment:
forEach()?

Reply:
I used the basic for loop because if I find a match, I want to stop immediately and return the whole _getInstructorTypeId() call. With forEach I'd have to complete the loop and check presence to know to exit.